apps/astarte_housekeeping/lib/astarte_housekeeping/config.ex

```diff
@@ -94,6 +109,37 @@ defmodule Astarte.Housekeeping.Config do
     end
   end
 
+  @doc """
+  Returns :ok if at least one of HOUSEKEEPING_ASTARTE_KEYSPACE_REPLICATION_FACTOR or HOUSEKEEPING_ASTARTE_KEYSPACE_NETWORK_REPLICATION_MAP is valid, based on HOUSEKEEPING_ASTARTE_KEYSPACE_REPLICATION_STRATEGY value.
+  """
+  def validate_astarte_replication! do
+    case astarte_keyspace_replication_strategy!() do
+      :simple -> validate_astarte_replication_factor!()
+      :network -> validate_astarte_replication_map!()
+      nil -> raise "Invalid replication strategy set for the astarte keyspace"
+    end
+  end
+
+  defp validate_astarte_replication_factor! do
+    case astarte_keyspace_replication_factor() do
+      {:ok, _} ->
+        :ok
```

Comment:
this still needs to validate the replication factor is not `nil` (ie: the cast was successful)

```suggestion
      {:ok, replication_factor} when replication_factor != nil ->
        :ok
```